src/main.ts

```diff
@@ -89,13 +92,87 @@ class NixInstallerAction {
       "diagnostic-endpoint",
     );
     this.trust_runner_user = action_input_bool("trust-runner-user");
-    this.correlation = process.env["STATE_correlation"];
+    this.correlation = correlation;
     this.nix_installer_url = resolve_nix_installer_url(
       this.platform,
       this.correlation,
     );
   }
 
+  async detectAndForceDockerShim(): Promise<void> {
+    // Detect if we're in a GHA runner which is Linux, doesn't have Systemd, and does have Docker.
+    // This is a common case in self-hosted runners, providers like [Namespace](https://namespace.so/),
+    // and especially GitHub Enterprise Server.
+    if (process.env.RUNNER_OS !== "Linux") {
+      if (!this.force_docker_shim) {
+        actions_core.warning(
```

Comment:
This seems like it should be an error to me...?

Reply:
I could make it a hard error. My thinking is if someone _forces_ it true, then who am I to say they shouldn't be allowed to keep both pieces? But I can also imagine a thing about being conservative about our inputs. wdyt?

Reply:
If it's going to be a warning, it'd be nice to make the "keep both pieces" more explicit somehow ("unsupported" is a lot softer than "this may break at any moment, we're not responsible if it blows up your cat or locks your key inside your car").

Reply:
I'm inclined to leave it as a warning and indicate we won't setup the docker shim if you're not on Linux.